backend/server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/storage/PluginsStorage.java

```diff
@@ -49,6 +49,15 @@ public PluginsStorage(Path connectionStorageRoot) {
     this.pluginReferencesFilePath = rootPath.resolve(PLUGIN_REFERENCES_PB);
   }
 
+  public boolean isValid() {
+    try {
+      rwLock.read(() -> ProtobufFileUtil.readFile(pluginReferencesFilePath, Sonarlint.PluginReferences.parser()));
```

Comment:
Shouldn't you also check if all plugin JARs are available?

Reply:
There is a log later when starting the analysis if one jar is missing/corrupted. I think it's better to proceed with the analysis at this point